orc/src/main/java/org/apache/iceberg/orc/OrcRowFilter.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.orc;
+
+import java.io.Serializable;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Type;
+
+/**
+ * [LINKEDIN ONLY]
+ * Represents a filter which can be applied to every row
+ * <p>
+ * Currently Row filter evaluation is only supported in Spark on unpartitioned tables for top level primitives
+ */
+public interface OrcRowFilter extends Serializable {
+
+  /**
+   * Returns a {@link Schema} which includes all fields required from the table {@link Schema} to evaluate this
+   * filter.
+   */
+  Schema requiredSchema();
+
+  /**
+   * Decides whether the row described by the provided field values should be kept or not.
+   * <p>
+   * The order of field values inside the values array is the same as the order of field ids returned by
+   * {@link #requiredSchema()}. The java class of the field value is dictated by {@link Type.TypeID#javaClass()}.
+   */
+  boolean shouldKeep(Object[] values);
```

Comment:
are these only top level columns. Should we call out as such?

Reply:
It is called out in the Javadoc of ORC row filter class.